datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/analytics/resolver/GetHighlightsResolver.java

```diff
@@ -37,44 +39,70 @@ public final List<Highlight> get(DataFetchingEnvironment environment) throws Exc
     }
   }
 
-  /**
-   * TODO: Config Driven Charts Instead of Hardcoded.
-   */
-  private List<Highlight> getHighlights() {
-    final List<Highlight> highlights = new ArrayList<>();
-
-    DateTime endDate = DateTime.now();
-    DateTime startDate = endDate.minusWeeks(1);
-    DateTime lastWeekStartDate = startDate.minusWeeks(1);
-    DateRange dateRange = new DateRange(String.valueOf(startDate.getMillis()), String.valueOf(endDate.getMillis()));
-    DateRange dateRangeLastWeek =
-        new DateRange(String.valueOf(lastWeekStartDate.getMillis()), String.valueOf(startDate.getMillis()));
-
-    // Highlight 1: The Highlights!
-    String title = "Weekly Active Users";
-    String eventType = "SearchEvent";
-
-    int weeklyActiveUsers =
-        _analyticsService.getHighlights(_analyticsService.getUsageIndexName(), Optional.of(dateRange),
-            ImmutableMap.of(), ImmutableMap.of(), Optional.of("browserId"));
-
-    int weeklyActiveUsersLastWeek =
-        _analyticsService.getHighlights(_analyticsService.getUsageIndexName(), Optional.of(dateRangeLastWeek),
-            ImmutableMap.of(), ImmutableMap.of(), Optional.of("browserId"));
+  private Highlight getTimeBasedHighlight(
+          String title,
```

Comment:
nit: can we mark everything that won't have its reference changed as `final`?

Reply:
sure